certbot/main.py

```diff
@@ -405,6 +406,53 @@ def _init_le_client(config, authenticator, installer):
 
     return client.Client(config, acc, authenticator, installer, acme=acme)
 
+def _cert_path_to_lineage(config):
+    """ If config.cert_path is defined, find an appropriate value for config.certname
+    by searching through available files in config.renewal_configs_dir, and finding
+    one with an appropriate value for 'fullchain'."""
+
+    for potential_conf in os.listdir(config.renewal_configs_dir):
+        lineage_name = storage.lineagename_for_filename(potential_conf)
+        renewal_file = storage.renewal_file_for_certname(config, lineage_name)
+        conf_fullchain = storage.fullchain_for_renewal_conf(renewal_file)
+
+        if conf_fullchain == config.cert_path[0]:
+            return lineage_name
+   
+def _delete_if_appropriate(config):
+    """Does the user want to delete their now-revoked certs?"""
+
+    if config.certname and config.cert_path:
+        # first, check if certname and cert_path imply the same certs
+        cert_path_implied_cert_name = _cert_path_to_lineage(config)
+        cert_path_implied_conf = storage.renewal_file_for_certname(config, cert_path_implied_cert_name)
+        cert_name_implied_conf = storage.renewal_file_for_certname(config, config.certname)
+ 
+        if cert_path_implied_conf != cert_name_implied_conf:
+            msg = "You specified conflicting values for --cert-path and --cert-name.\
+                   For --cert-path you specified $0, which implies the following conf file: $1.\
+                   For --cert-name you specified $2, which implies the following conf file: $3."\
+                   .format(config.cert_path, cert_path_implied_conf,
+                           config.certname, cert_name_implied_conf)
+            choices = [cert_path_implied_conf, cert_name_implied_conf]
+            display = zope.component.getUtility(interfaces.IDisplay)
+            code, index = display.menu("Which did you mean to select?",
+                    choices, ok_label="Select", force_interactive=True)
+
+            if code != display_util.OK or not index in range(0, len(choices)):
+                raise errors.Error("User ended interaction.")
+
+            if index == 0:
+                config.certname = cert_path_implied_cert_name 
+            else:
+                cert_path = storage.fullchain_for_renewal_conf(cert_name_implied_conf)
+                with open(cert_path) as f:
+                    config.cert_path = (cert_path, f.read()) 
+
+    else: # if only config.cert_path was specified
+        config.cert_name = _cert_path_to_lineage(config)
```

Comment:
If `_cert_path_to_lineage` doesn't find a match, we should probably print a message about how we were unable to find a relevant cert to delete.